File: Bai_64/main.py
def find_index(a):
    tong = sum(a)
    mang1 = a[0]
    k = 0
    for i in range(1, len(a)):
        if abs(tong - 2 * mang1) >= abs(tong - 2 * (mang1 + a[i])):
            k = i
        mang1 += a[i]
    return k

if __name__ == "__main__":
    n = int(input())
    a = [int(input()) for _ in range(n)]
    print(find_index(a))

File: Bai_64/main.cpp
#include <iostream>
using namespace std;

int find_index(int a[], int &n){
    int tong = 0;
    for(int i = 0; i < n; i++)
        tong += a[i];
    int mang1 = a[0];
    int k = 0;
    for(int i = 1; i < n; i++){
        if(abs(tong - 2 * mang1) >= abs(tong - 2 * (mang1 + a[i])))
            k = i;
        mang1 += a[i];
    }
    return k;
}

int main(){
    int a[100001];
    int n; cin >> n;
    for(int i = 0; i < n; i++)
        cin >> a[i];
    cout << find_index(a, n) << endl;
    return 0;
}